src/main/java/org/terasology/splash/AbstractSplashScreen.java

```diff
@@ -96,19 +100,30 @@ public void close() {
 
     @Override
     public void post(String message) {
-        if (message != null) {
+        if (message == null) {
+            return;
+        }
+
+        synchronized (lock) {
             messageQueue.add(message);
```

Comment:
This will actually throw an exception if the queue is full. You might want messageQueue.offer(message) - except that will cause the new message to be dropped if the queue is full. The desired behavior is probably to drop the oldest message, not the newest one.